javascript/modules/client/src/client-addressList.ts

```diff
@@ -70,173 +23,25 @@ const PLUGIN_ID = "0x1234567890123456789012345678901234567890";
-     * @return {*}  {Promise<IPluginSettings>}
-     * @memberof ClientAddressList
-     */
-    getSettings: (pluginAddress: string): Promise<IPluginSettings | null> =>
-      this._getSettings(pluginAddress),
-  };
-  encoding = {
-    /**
-     * Computes the parameters to be given when creating a proposal that updates the governance configuration
-     *
-     * @param {string} pluginAddress
-     * @param {IPluginSettings} params
-     * @return {*}  {DaoAction}
-     * @memberof ClientAddressList
-     */
-    updatePluginSettingsAction: (
-      pluginAddress: string,
-      params: IPluginSettings,
-    ): DaoAction =>
-      this._buildUpdatePluginSettingsAction(pluginAddress, params),
-
-    /**
-     * Computes the parameters to be given when creating a proposal that adds addresses to address list
-     *
-     * @param pluginAddress
-     * @param members
-     * @returns {*} {DaoAction}
-     */
-    addMembersAction: (pluginAddress: string, members: string[]): DaoAction =>
-      this._buildAddMembersAction(pluginAddress, members),
-
-    /**
-     * Computes the parameters to be given when creating a proposal that removes addresses from the address list
-     *
-     * @param pluginAddress
-     * @param members
-     * @returns {*} {DaoAction}
-     */
-    removeMembersAction: (
-      pluginAddress: string,
-      members: string[],
-    ): DaoAction => this._buildRemoveMembersAction(pluginAddress, members),
-  };
-  decoding = {
-    /**
-     * Decodes a dao metadata from an encoded update metadata action
-     *
-     * @param {Uint8Array} data
-     * @return {*}  {IPluginSettings}
-     * @memberof ClientAddressList
-     */
-    updatePluginSettingsAction: (data: Uint8Array): IPluginSettings =>
-      decodeUpdatePluginSettingsAction(data),
-
-    /**
-     * Decodes a list of addresses from an encoded add members action
-     *
-     * @param data
-     * @returns {*}  {string[]}
-     */
-    addMembersAction: (data: Uint8Array): string[] =>
-      decodeAddMemebersAction(data),
-
-    /**
-     * Decodes a list of addresses from an encoded remove members action
-     *
-     * @param data
-     * @returns {*}  {string[]}
-     */
-    removeMembersAction: (data: Uint8Array): string[] =>
-      decodeRemoveMemebersAction(data),
-
-    /**
-     * Returns the decoded function info given the encoded data of an action
-     *
-     * @param {Uint8Array} data
-     * @return {*}  {IInterfaceParams | null}
-     * @memberof ClientAddressList
-     */
-    findInterface: (data: Uint8Array): IInterfaceParams | null =>
-      this._findInterfaceParams(data),
-  };
   static encoding = {
     /**
      * Computes the parameters to be given when creating the DAO,
      * so that the plugin is configured
      *
-     * @param {IErc20PluginInstall} params
-     * @return {*}  {FactoryInitParams}
+     * @param {IAddressListPluginInstall} params
+     * @return {*}  {IPluginInstallItem}
      * @memberof ClientAddressList
      */
```

Comment:
Why not importing the static bit from the module file as well?

Reply:
My point here is that if this is a static function, then we could have the implementation as a standalone function implemented alongside the class. 

Something like:
```ts
import { getPluginInstallItem } from "....";

class Client {
  // ...

  static encoding = {
    getPluginInstallItem: (...params) => getPluginInstallItem(...params)
  }
}
```